examples/oneapi/cpp/source/kmeans/kmeans_lloyd_dense_batch.cpp

```diff
@@ -0,0 +1,67 @@
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "example_util/utils.hpp"
+#include "oneapi/dal/algo/kmeans.hpp"
+
+using namespace oneapi;
+
+int main(int argc, char const *argv[]) {
+  constexpr std::int64_t row_count = 8;
+  constexpr std::int64_t column_count = 7;
+
+  const float data[] = {1.f, 2.f,  3.f, 4.f, 5.f, 6.f,  -5.f, 1.f, -1.f, 0.f,
+                        3.f, 1.f,  2.f, 3.f, 4.f, 5.f,  6.f,  1.f, 0.f,  0.f,
+                        0.f, 1.f,  2.f, 5.f, 2.f, 9.f,  3.f,  2.f, -4.f, 3.f,
+                        0.f, 4.f,  2.f, 7.f, 5.f, 4.f,  2.f,  0.f, -4.f, 0.f,
+                        3.f, -8.f, 2.f, 5.f, 5.f, -6.f, 3.f,  0.f, -9.f, 3.f,
+                        1.f, -3.f, 3.f, 5.f, 1.f, 7.f};
+
+  const float initial_centroids[] = {1.f, 2.f, 3.f, 4.f, 5.f, 6.f, -5.f,
+                                     1.f, 2.f, 5.f, 2.f, 9.f, 3.f, 2.f};
+
+  const auto data_table = dal::homogen_table{row_count, column_count, data};
+  const auto initial_centroids_table =
+      dal::homogen_table{1, column_count, initial_centroids};
+
+  const auto kmeans_desc = dal::kmeans::descriptor<>()
+                               .set_cluster_count(2)
+                               .set_max_iteration_count(5)
+                               .set_accuracy_threshold(1.0);
```

Comment:
Why do you have only 5 iterations and a threshold of more than 1? Judging by the objective function, you do not solve the problem well. Maybe tighten the parameters?